### ISSUE TITLE
GPT2 -- fail to build engine with "past_key_values" (see https://huggingface.co/docs/transformers/model_doc/gpt2)

### ISSUE THREAD

**Issue by kobzaond** [NVIDIA/TensorRT#1808]
## Description

I converted the Huggingface GPT2 Pytorch model into onnx format with support of the past-key_values: i.e. the input contains "input_ids, attention_mask" and keys and values for each of the attention block, it outputs logits and computed all keys and values, aka "cache". Using the cache speeds generation on my GPU significantly.

Afterward I tried to convert this model to trt engine: when I use constant shapes (MIN=OPT=MAX) it is alright, however when I try to use dynamic shapes (which is necessary for text generation),I receive following error:

```
[E] 4: [shapeCompiler.cpp::evaluateShapeChecks::832] Error Code 4: Internal Error (kOPT values for profile 0 violate shape constraints: condition '==' violated. IAssertionLayer (Unnamed Layer* 5) [Assertion]: condition[0] is false: (EQUAL (# 0 (SHAPE input_ids)) (# 0 (SHAPE past_key_values.0.key))). For input: 'input_ids' all named dimensions that share the same name must be equal. Note: Named dimensions were present on the following axes: 0 (name: 'batch'), 0 (name: 'batch'))

```

*tried with 'gpt2' model, the past key values are of shape [beam, 12, seq_length, 64]
* conversion is done using Python API

## Environment

**TensorRT Version**: 8.2.1.8
**NVIDIA GPU**: TESLA T4
**NVIDIA Driver Version**: 
**CUDA Version**: 11.5
**CUDNN Version**: 8.2
**Operating System**: ubuntu 18:04
**Python Version (if applicable)**: python 3.9
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 1.10.2+cu113
**Baremetal or Container (if so, version)**:

**Comment by ttyio** [comment]
Hi @kobzaond , Is the `batch` dim of the inputs has the same min/opt/max? thanks

**Comment by kobzaond** [comment]
the batch size was fixed (same for min/opt/max), however the 'length' dimensionality varies (it must be dynamic for generation purpose).

**Comment by kobzaond** [comment]
Hello, I tried to use the trtexec command and the result is the same:

```
[03/18/2022-17:15:05] [I] Finish parsing network model
[03/18/2022-17:15:05] [W] [TRT] Calibrator is not being used. Users must provide dynamic range for all tensors that are not Int32 or Bool.
[03/18/2022-17:15:09] [I] [TRT] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +508, GPU +222, now: CPU 1997, GPU 502 (MiB)
[03/18/2022-17:15:10] [I] [TRT] [MemUsageChange] Init cuDNN: CPU +432, GPU +172, now: CPU 2429, GPU 674 (MiB)
[03/18/2022-17:15:10] [I] [TRT] Local timing cache in use. Profiling results in this builder pass will not be stored.
[03/18/2022-17:15:10] [E] Error[4]: [shapeCompiler.cpp::evaluateShapeChecks::832] Error Code 4: Internal Error (kOPT values for profile 0 violate shape constraints: broadcast with incompatible dimensions. Add_162: dimensions not compatible for elementwise)
[03/18/2022-17:15:10] [E] Error[2]: [builder.cpp::buildSerializedNetwork::609] Error Code 2: Internal Error (Assertion enginePtr != nullptr failed. )
[03/18/2022-17:15:10] [E] Engine could not be created from network
[03/18/2022-17:15:10] [E] Building engine failed
[03/18/2022-17:15:10] [E] Failed to create engine from model.
[03/18/2022-17:15:10] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8201] # /usr/src/tensorrt/bin/trtexec --onnx=model-support-cache.onnx --workspace=8000 --sparsity=enable --best --minShapes=input_ids:5x1,attention_mask:5x1,past_key_values.0.key:5x12x1x64,past_key_values.1.key:5x12x1x64,past_key_values.2.key:5x12x1x64,past_key_values.3.key:5x12x1x64,past_key_values.4.key:5x12x1x64,past_key_values.5.key:5x12x1x64,past_key_values.6.key:5x12x1x64,past_key_values.7.key:5x12x1x64,past_key_values.8.key:5x12x1x64,past_key_values.9.key:5x12x1x64,past_key_values.10.key:5x12x1x64,past_key_values.11.key:5x12x1x64,past_key_values.0.value:5x12x1x64,past_key_values.1.value:5x12x1x64,past_key_values.2.value:5x12x1x64,past_key_values.3.value:5x12x1x64,past_key_values.4.value:5x12x1x64,past_key_values.5.value:5x12x1x64,past_key_values.6.value:5x12x1x64,past_key_values.7.value:5x12x1x64,past_key_values.8.value:5x12x1x64,past_key_values.9.value:5x12x1x64,past_key_values.10.value:5x12x1x64,past_key_values.11.value:5x12x1x64 --optShapes=input_ids:5x50,attention_mask:5x50,past_key_values.0.key:5x12x50x64,past_key_values.1.key:5x12x50x64,past_key_values.2.key:5x12x50x64,past_key_values.3.key:5x12x50x64,past_key_values.4.key:5x12x50x64,past_key_values.5.key:5x12x50x64,past_key_values.6.key:5x12x50x64,past_key_values.7.key:5x12x50x64,past_key_values.8.key:5x12x50x64,past_key_values.9.key:5x12x50x64,past_key_values.10.key:5x12x50x64,past_key_values.11.key:5x12x50x64,past_key_values.0.value:5x12x50x64,past_key_values.1.value:5x12x50x64,past_key_values.2.value:5x12x50x64,past_key_values.3.value:5x12x50x64,past_key_values.4.value:5x12x50x64,past_key_values.5.value:5x12x50x64,past_key_values.6.value:5x12x50x64,past_key_values.7.value:5x12x50x64,past_key_values.8.value:5x12x50x64,past_key_values.9.value:5x12x50x64,past_key_values.10.value:5x12x50x64,past_key_values.11.value:5x12x50x64 --maxShapes=input_ids:5x50,attention_mask:5x50,past_key_values.0.key:5x12x50x64,past_key_values.1.key:5x12x50x64,past_key_values.2.key:5x12x50x64,past_key_values.3.key:5x12x50x64,past_key_values.4.key:5x12x50x64,past_key_values.5.key:5x12x50x64,past_key_values.6.key:5x12x50x64,past_key_values.7.key:5x12x50x64,past_key_values.8.key:5x12x50x64,past_key_values.9.key:5x12x50x64,past_key_values.10.key:5x12x50x64,past_key_values.11.key:5x12x50x64,past_key_values.0.value:5x12x50x64,past_key_values.1.value:5x12x50x64,past_key_values.2.value:5x12x50x64,past_key_values.3.value:5x12x50x64,past_key_values.4.value:5x12x50x64,past_key_values.5.value:5x12x50x64,past_key_values.6.value:5x12x50x64,past_key_values.7.value:5x12x50x64,past_key_values.8.value:5x12x50x64,past_key_values.9.value:5x12x50x64,past_key_values.10.value:5x12x50x64,past_key_values.11.value:5x12x50x64
```

batch size=5; min length=1, opt and max length =50, the other dimensions are fixed; model: gpt2 with cache

**Comment by nvpohanh** [comment]
@kobzaond Could you share the ONNX file so that we can repro and debug? Thanks

**Comment by nvpohanh** [comment]
Closing due to >14 days without activity. Please feel free to reopen if the issue still exists. Thanks
